### PR TITLE
[MAINTENANCE] Type hint cleanup in usage statistics

### PR DESCRIPTION
**Pull request by cdkini**
- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/core/usage_statistics/usage_statistics.py [hunk 1]
```diff
@@ -222,15 +222,15 @@ def get_usage_statistics_handler(args_array: list) -> Optional[UsageStatisticsHa
...
 
 def usage_statistics_enabled_method(
-    func: Optional[Callable] = None,
-    event_name: Optional[UsageStatsEvents] = None,
-    args_payload_fn: Optional[Callable] = None,
-    result_payload_fn: Optional[Callable] = None,
+    func: Callable | None = None,
+    event_name: UsageStatsEvents | None = None,
+    args_payload_fn: Callable | None = None,
+    result_payload_fn: Callable | None = None,
```
Comment (Kilo59) [inline great_expectations/core/usage_statistics/usage_statistics.py:233]: Is there anything else about the callable that we can declare here?
Like what they are expected to return or the params they accept?

**Review by Kilo59 — APPROVED** [review]
❤️

File: great_expectations/core/usage_statistics/usage_statistics.py (hunk 1 above)
Comment (cdkini) [reply great_expectations/core/usage_statistics/usage_statistics.py:233]: unfortunately its a super variable input list (the fn in question here is any method we decorate)

Comment (cdkini) [reply great_expectations/core/usage_statistics/usage_statistics.py:233]: if you look at the context, we have a number of varied usage stats decorated methods

**Pull request merged**